### PR TITLE
Fixes For Parameter Metadata Format Changes

### PR DESCRIPTION
**Pull request by simpsont-oci**
This should fix data loading for server-side parameter metadata format changes. Also updates to Node 16, updates module versions, resolves svelte / typescript issues, and updates mocks and and snapshots for the new format. The only 'unresolved' issues that I could see were a svelte-check complaint about `document.execCommand('Copy')` which seemed like it might be wise to preserve for compatibility with older browsers and some npm audit complaints that are occurring because c3 is no longer actively maintained and the most recent release relies on versions of d3 with known security issues.

### CODE REVIEW

File: ui/src/AppCharting/chart-data-extractor.ts [hunk 1]
```diff
@@ -134,20 +153,42 @@ export async function getChartDataByTimestamp(
...
+      let value: number = MISSING_VALUE;
       const benchmark = benchmarkMap[timestamp.key];
-      const scenarioSizeRecords = benchmark && benchmark[scenario];
-      if (benchmark && scenarioSizeRecords) {
-        const plotTypeRecords =
-          scenarioSizeRecords[isFan ? `${dataName}_${serverCount}` : dataName];
-        if (plotTypeRecords) {
-          const plotStatisctics = plotTypeRecords[plotType];
-          if (plotStatisctics) value = plotStatisctics[statName];
+      if (benchmark) {
```
Comment (eahrold) [inline ui/src/AppCharting/chart-data-extractor.ts:158]: oh man... there's probably some special handling here but it sure looks a whole lot like the handler above.

File: ui/src/AppSharing/AppSharing.svelte [hunk 2]
```diff
@@ -53,22 +53,31 @@
 </script>
 
 <div class="sharing">
-  <button on:click={() => (visible = true)}>Embed and Share Info</button>
+  <button on:click={() => (visible = true)} on:keypress={() => (visible = true)}
```
Comment (eahrold) [inline ui/src/AppSharing/AppSharing.svelte:56]: not sure if this is the desired behavior, but you may want to narrow the key that triggers this event to something like "Enter"

https://www.w3schools.com/howto/howto_js_trigger_button_enter.asp

**Review by eahrold — APPROVED** [review]
Looks good.  A couple of suggestions, there may be some ways to DRY out the benchmark map, but will leave that to you if it's worth the time.

File: ui/src/AppCharting/chart-data-extractor.ts (hunk 1 above)
Comment (simpsont-oci) [reply ui/src/AppCharting/chart-data-extractor.ts:158]: I agree, there was already some duplication in the handlers and I probably added to it a bit. But sometime soon there will be an additional follow-on task to make more extensive use of the new parameter metadata. Those changes will likely require more extensive changes to the functions that extract data from the benchmark map, so I think it's OK to leave it for now and I'll make a note to try to reduce duplication as part of the next task.

File: ui/src/AppSharing/AppSharing.svelte (hunk 2 above)
Comment (simpsont-oci) [reply ui/src/AppSharing/AppSharing.svelte:56]: I'm pretty sure this one was from svelte-check asking for an "on:keypress" to go along with the "on:click" so I just copied the handler. Limiting it to "Enter" makes sense.

**Pull request merged**